xls/passes/rematerialization_pass.cc

```diff
@@ -388,6 +434,87 @@ FindRematerializationOpportunities(
   return result;
 }
 
+// TODO(zihao): replace `FindRematerializationOpportunitiesAtNode` later
+absl::StatusOr<RematEdgeOpportunity>
+FindRematerializationOpportunitiesAtEdge(
+    Node* src, Node* dst, FunctionBase* f,
+    const absl::flat_hash_map<Node*, int64_t>& schedule,
+    const InverseSchedule& inverse_schedule) {
+  int64_t dst_stage_num = schedule.at(dst);
+  absl::flat_hash_set<Node*> available;
+  for (Node* node : inverse_schedule[dst_stage_num]) {
```

Comment:
I realize now that this is the wrong notion of an available value. A value is available at stage k if it is defined in stage a and its last use is in stage b and `a ≤ k ≤ b`.

Reply:
Great idea about the new notion of availability! It can reduce the size of chunks.